src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -377,3 +391,56 @@ SupportFunctionForColumn(Var *partitionColumn, Oid accessMethodId,
 
 	return supportFunctionOid;
 }
+
+
+/*
+ * TableHasData function checks whether given undistributed table contains any row and
+ * returns true if there is any data. This function is only for undistributed tables and
+ * should not be called for distributed tables.
+ */
+static bool
+TalbeHasData(Oid tableId)
+{
+	Oid schemaId = get_rel_namespace(tableId);
+	char *schemaName = get_namespace_name(schemaId);
+	char *tableName = get_rel_name(tableId);
+	char *tableQualifiedName = quote_qualified_identifier(schemaName, tableName);
+
+	int spiConnectionResult = 0;
+	int spiQueryResult = 0;
+	StringInfo selectExistQueryString = makeStringInfo();
+
+	HeapTuple tuple = NULL;
+	Datum hasDataDatum = 0;
+	bool hasData = false;
+	bool isNull = false;
+
+	int rowId = 0;
+	int attributeId = 1;
+
+	spiConnectionResult = SPI_connect();
+	if (spiConnectionResult != SPI_OK_CONNECT)
+	{
+		ereport(ERROR, (errmsg("could not connect to SPI manager")));
+	}
+
+	appendStringInfo(selectExistQueryString, SELECT_EXIST_QUERY, tableQualifiedName);
+
+	spiQueryResult = SPI_execute(selectExistQueryString->data, true, 0);
```

Comment:
I think instead of using `true`, it is better to declare a variable (i.e, `readOnly = true`) and pass it here.